src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1366,7 +1366,7 @@ public static bool IsPartial(this TypeSymbol type)
 
         public static bool HasFileLocalTypes(this TypeSymbol type)
         {
-            var foundType = type.VisitType(predicate: (type, _, _) => type is SourceMemberContainerTypeSymbol { IsFileLocal: true }, arg: (object?)null);
+            var foundType = type.VisitType(predicate: (type, _, _) => type.OriginalDefinition is SourceMemberContainerTypeSymbol { IsFileLocal: true }, arg: (object?)null);
```

Comment:
What kind of symbol was causing a problem, was it `SubstitutedNamedTypeSymbol`? If so, shouldn't `SubstitutedNamedTypeSymbol.IsFileLocal` already defer to the original definition?

Reply:
Per offline discussion we'd like to try introducing `internal abstract bool NamedTypeSymbol.IsFileLocal { get; }` and using it where applicable.